app/helpers/Evaluation/EvaluationPointsLoader.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Helpers;
+
+use App\Model\Entity\Submission;
+use App\Model\Entity\SolutionEvaluation;
+use App\Model\Entity\ReferenceSolutionEvaluation;
+use App\Helpers\JobConfig\Storage as JobConfigStorage;
+use App\Helpers\EvaluationResults\EvaluationResults;
+use App\Helpers\EvaluationResults\Loader as EvaluationResultsLoader;
+
+
+/**
+ * Load points for given solution evaluation.
+ */
+class EvaluationPointsLoader {
+
+  /**
+   * @var ScoreCalculatorAccessor
+   */
+  private $calculators;
+
+  /**
+   * EvaluationPointsLoader constructor.
+   * @param ScoreCalculatorAccessor $calculators
+   */
+  public function __construct(ScoreCalculatorAccessor $calculators) {
+    $this->calculators = $calculators;
+  }
+
+
+  /**
+   * Set score and points to given evaluation of student submission.
+   * @param SolutionEvaluation $evaluation
+   */
+  public function setStudentScoreAndPoints(SolutionEvaluation $evaluation) {
+    $this->setStudentScore($evaluation);
+    $this->setStudentPoints($evaluation);
+  }
+
+  /**
+   * Set score to evaluation of student submission.
+   * @param SolutionEvaluation $evaluation
+   */
+  private function setStudentScore(SolutionEvaluation $evaluation) {
+    if ($evaluation === null || $evaluation->getSubmission() === null) {
+      // not a student submission
+      return;
+    }
+
+    $score = 0;
+    $submission = $evaluation->getSubmission();
+    $calculator = $this->calculators->getCalculator($submission->getAssignment()->getScoreCalculator());
+
+    // calculate scores for all tests
+    $scores = [];
+    foreach ($evaluation->getTestResults() as $testResult) {
+      $scores[$testResult->getTestName()] = $testResult->getScore();
+    }
+
+    // calculate percentual score of whole submission
+    if ($calculator !== NULL && !$evaluation->getInitFailed()) {
+      $score = $calculator->computeScore($submission->getAssignment()->getScoreConfig(), $scores);
+    }
+
+    // ... and set results
+    $evaluation->setScore($score);
+  }
+
+  /**
+   * Set points to evaluation of student submission.
+   * @note Score has to be calculated before call of this function.
+   * @param SolutionEvaluation $evaluation
+   */
+  public function setStudentPoints(SolutionEvaluation $evaluation) {
+    if ($evaluation === null || $evaluation->getSubmission() === null) {
+      // not a student submission
+      return;
+    }
+
+    // setup
+    $submission = $evaluation->getSubmission();
+    $maxPoints = $submission->getMaxPoints();
+
+    // calculate points from the score
+    $points = floor($evaluation->getScore() * $maxPoints);
```

Comment:
Would not round() be better?

Reply:
Not sure, this is original code, which I did not write.

Reply:
Because 0.95 is almost 1, but it is not fully correct, so floor reflects this better.

Reply:
That may be true, but students receive the best grade even if their test/oral exam/... is not completely correct. I mean if you have like 20 tests (weighted evenly) and maxPoint == 3, it would be nice to assign 3 points even for 19 correct tests. Just my opinion.

Reply:
Let's discuss this tomorrow, but we think that CodEx test are designed in a way that even one failed test means a penalty. If there is a `round()` function, it would be common that students will receive all of the points for suboptimal solutions and for example Mr. Jezek will complain instantly. We suppose, that each exercise have 10 - 20 tests and maximum of 10 points.

Reply:
I completely agree with @SemaiCZE here - if some tests are not necessary for a 100% score, it must be stated explicitly (which I think is not possible now, but that's another issue).